OpenEdXMobile/src/main/java/org/edx/mobile/base/FindCoursesBaseActivity.java

```diff
@@ -351,17 +381,35 @@ public void onPageFinished() {
         }
 
         @Override
-        public void onPageLoadError(WebView view, int errorCode, String description, String failingUrl) {
+        public void onPageLoadError(WebView view, int errorCode, String description,
+                                    String failingUrl) {
             isWebViewLoaded = false;
-            showOfflineMessage();
+            switch (errorCode) {
+                case WebViewClient.ERROR_HOST_LOOKUP:
```

Comment:
This check doesn't look good to me
As per documentation: 
https://developer.android.com/reference/android/webkit/WebViewClient.html

```
ERROR_HOST_LOOKUP
Server or proxy hostname lookup failed
```
This error belongs to hostname failure majorly and in this case we shall show a `'Interner connection'` error message as passing `IOException()` which is not right.

We suppose to apply an internet connectivity check before loading url and can also check on top of `onPageLoadError` callback if feasible which will fulfill our problem.